### PR TITLE
feat!: Extract enums to top level and use them in Js interface

### PR DESCRIPTION
**Pull request by enricocolasante**
All enums are now top level types and exported to JS.

We opt-in to @JsExport via setting to not require the @OptIn annotation in combination with @JsExport

### CODE REVIEW

File: src/commonMain/kotlin/org/hisp/dhis/rules/models/EnrollmentStatus.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.hisp.dhis.rules.models
+
+import kotlin.js.ExperimentalJsExport
+import kotlin.js.JsExport
+
+@JsExport
+@OptIn(ExperimentalJsExport::class)
+enum class EnrollmentStatus {
```
Comment (vgarciabnz) [inline src/commonMain/kotlin/org/hisp/dhis/rules/models/EnrollmentStatus.kt]: This name could to collide with `EnrollmentStatus` enum defined in backend and android (at least in Android for sure), which would force to use the full package name when used in the same file. Maybe `RuleEnrollmentStatus`? Or is it unnecessary long?

File: src/commonMain/kotlin/org/hisp/dhis/rules/models/EventStatus.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.hisp.dhis.rules.models
+
+import kotlin.js.ExperimentalJsExport
+import kotlin.js.JsExport
+
+@JsExport
+@OptIn(ExperimentalJsExport::class)
+enum class EventStatus {
```
Comment (vgarciabnz) [inline src/commonMain/kotlin/org/hisp/dhis/rules/models/EventStatus.kt]: Same comment as for EnrollmentStatus

File: src/commonMain/kotlin/org/hisp/dhis/rules/models/EnrollmentStatus.kt (hunk 1 above)
Comment (enricocolasante) [reply src/commonMain/kotlin/org/hisp/dhis/rules/models/EnrollmentStatus.kt]: Done

File: src/commonMain/kotlin/org/hisp/dhis/rules/models/EventStatus.kt (hunk 2 above)
Comment (enricocolasante) [reply src/commonMain/kotlin/org/hisp/dhis/rules/models/EventStatus.kt]: Done

**Review by jbee — APPROVED** [review]
That was really straight forward :) 

I think overall this was a good decision that made the code easy to grasp and got rid of some boilerplate/mapping code at the same time.

**Review by vgarciabnz — APPROVED** [review]

**Pull request merged**